src/Asset/ManifestLookup.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Asset;
+
+/**
+ * Looks up values in Encore's `manifest.json` and returns them.
+ *
+ * @final
+ */
+class ManifestLookup
+{
+    private $manifestPath;
+    private $manifestData;
+
+    /**
+     * @param string $manifestPath Absolute path to the manifest.json file
+     */
+    public function __construct(string $manifestPath)
+    {
+        $this->manifestPath = $manifestPath;
+    }
+
+    public function getManifestPath(string $path): ?string
+    {
+        if (null === $this->manifestData) {
+            if (!file_exists($this->manifestPath)) {
+                throw new \RuntimeException(sprintf('Asset manifest file "%s" does not exist.', $this->manifestPath));
+            }
+
+            $this->manifestData = json_decode(file_get_contents($this->manifestPath), true);
+            if (0 < json_last_error()) {
```

Comment:
What about making this `JSON_ERROR_NONE !== json_last_error()`? It would make it more clear what is happening

Reply:
I agree here